plugins/inputs/redis/redis.go

```diff
@@ -321,6 +327,48 @@ func gatherKeyspaceLine(
 	}
 }
 
+// Parse the special Replication line
+// Example:
+//     slave0:ip=127.0.0.1,port=7379,state=online,offset=4556468,lag=0
+// This line will only be visible when a node has a replica attached.
+func gatherReplicationLine(
+	name string,
+	line string,
+	acc telegraf.Accumulator,
+	global_tags map[string]string,
+) {
+	fields := make(map[string]interface{})
+	tags := make(map[string]string)
+	for k, v := range global_tags {
+		tags[k] = v
+	}
+
+	tags["replica_id"] = strings.TrimLeft(name, "slave")
+	// change our role for just the replicas (slaves)
+	tags["replication_role"] = "replica"
```

Comment:
I think we should set this to `slave`, since connecting the plugin directly to the slave would currently set `role=slave`.